compiler/src/dotty/tools/backend/jvm/DottyBackendInterface.scala

```diff
@@ -251,14 +251,12 @@ class DottyBackendInterface(outputDirectory: AbstractFile, val superCallsMap: Ma
           case StringTag =>
             assert(const.value != null, const) // TODO this invariant isn't documented in `case class Constant`
             av.visit(name, const.stringValue) // `stringValue` special-cases null, but that execution path isn't exercised for a const with StringTag
-          case ClazzTag => av.visit(name, const.typeValue.toTypeKind(bcodeStore)(innerClasesStore).toASMType)
+          case ClazzTag => av.visit(name, TypeErasure.erasure(const.typeValue).toTypeKind(bcodeStore)(innerClasesStore).toASMType)
```

Comment:
Erasure should have erased the `typeValue`s of `Constant`s. It's not good to keep a non-erased type in the `Constant`s after erasure.

Reply:
Annotation trees are never transformed so they require some special care like this.

Reply:
Nevertheless, scalac doesn't do this in the codegen for `Literal`s:
https://github.com/scala/scala/blob/cb31fee193b1d9031a59d1b9fccb18aa7b4404c0/src/compiler/scala/tools/nsc/backend/jvm/BCodeBodyBuilder.scala#L456-L460
Maybe we need to handle annotation trees specially, but for regular `Literal`s in the code I believe they should be erased.

Reply:
Yes, regular literals are fine, in dotty like in scalac they're handled by genConstant which doesn't call erasure.